### DISCUSSION TITLE
‘Add to Home Screen’ is broken because `/static/manifest.json` sets `start_url`/`homepage` to `.`

### DISCUSSION THREAD

**Discussion by huw** [community/community#31578] [Codespaces] [answered]
Hey folks,

Codespaces broke again on my iPad today. When opening from Safari, everything is fine, but when opening the PWA from my home screen I get a blank `Not Found` page.

Didn’t take long to dig around and find the culprit. In dev tools, I could see that the PWA was trying to open `blahblahblah.github.dev/static/` instead of the root path. Since moving to your new CDN at the `/static/` path, you’re hosting `manifest.json` at `/static/manifest.json`, which reads like:

```json
{
	"short_name": "Codespaces",
	"name": "Codespaces",
	"icons": [
		{
			"src": "/static/favicon.ico",
			"sizes": "64x64 32x32 24x24 16x16",
			"type": "image/x-icon"
		},
		{
			"src": "/static/gh-192.png",
			"sizes": "192x192",
			"type": "image/png"
		},
		{
			"src": "/static/gh-512.png",
			"sizes": "512x512",
			"type": "image/png"
		}
	],
	"start_url": ".",
	"homepage": ".",
	"display": "fullscreen",
	"theme_color": "#2C2C2D",
	"background_color": "#1B1B1B"
}
```

Unfortunately when iPadOS reads this it resolves `start_url` and `homepage` (and the icons, for that matter) relative to where `manifest.json` is hosted, so it tries to open `/static` instead of `/`.

This occurs for Codespaces hosted in any region, and I encountered the error while browsing locally (Sydney) and over a VPN to the U.S. If it helps, I rely on Codespaces on iPad for a huge percentage of my work and it’s frustrating that it’s seemingly not tested on this platform :(

**Comment by fara-nak** [comment]
sorry you were facing this problem, I have created an issue on our side to investigate and fix this issue.

**Comment by greggroth** [comment, marked as answer]
~Can you tell us more about how you're using Codespaces? Are you attempting to have a long-running web process in your codespace that serves the `manifest.json` file?~

Apologies! I've talked with some folks more familiar with the PWA and I learned there was a bug that matches the behavior you saw that was fixed last night. Can you confirm if things are working for you know or if you still have issues?

**Reply by huw** [reply]
Thanks—it’s fixed! I had to delete & recreate my Codespace but the new one works :)
